syntax = "proto3";

package NAlice;

option go_package = "a.yandex-team.ru/alice/protos/data/device";
option java_package = "ru.yandex.alice.protos.data.device";
option java_outer_classname = "Info";

import "mapreduce/yt/interface/protos/extension.proto";

enum EUserDeviceType {
    UnknownDeviceType = 0;
    LightDeviceType = 1;
    SocketDeviceType = 2;
    SwitchDeviceType = 3;
    HubDeviceType = 4;

    PurifierDeviceType = 5;
    HumidifierDeviceType = 6;
    VacuumCleanerDeviceType = 7;

    CookingDeviceType = 8;
    KettleDeviceType = 9;
    CoffeeMakerDeviceType = 10;

    ThermostatDeviceType = 11;
    AcDeviceType = 12;

    MediaDeviceDeviceType = 13;
    TvDeviceDeviceType = 14;
    ReceiverDeviceType = 15;
    TvBoxDeviceType = 16;

    WashingMachineDeviceType = 17;

    OpenableDeviceType = 18;
    CurtainDeviceType = 19;

    SmartSpeakerDeviceType = 20;
    YandexStationDeviceType = 21;
    YandexStation2DeviceType = 22;
    YandexStationMiniDeviceType = 23;
    DexpSmartBoxDeviceType = 24;
    IrbisADeviceType = 25;
    ElariSmartBeatDeviceType = 26;
    LGXBoomDeviceType = 27;
    JetSmartMusicDeviceType = 28;
    PrestigioSmartMateDeviceType = 29;
    DigmaDiHomeDeviceType = 30;
    JBLLinkPortableDeviceType = 31;
    JBLLinkMusicDeviceType = 32;

    YandexModuleDeviceType = 33;

    RemoteCarDeviceType = 34;

    OtherDeviceType = 35;

    YandexStationMini2DeviceType = 36;

    DishwasherDeviceType = 37;

    MulticookerDeviceType = 38;
    RefrigeratorDeviceType = 39;
    FanDeviceType = 40;
    IronDeviceType = 41;

    SensorDeviceType = 42;

    YandexModule2DeviceType = 43;
    YandexStationMicroDeviceType = 44;

    PetFeederDeviceType = 45;

    YandexStationCentaurDeviceType = 46;

    LightCeilingDeviceType = 47;
    LightLampDeviceType = 48;
    LightStripDeviceType = 49;

    YandexStationMidiDeviceType = 50;
    YandexStationMini2NoClockDeviceType = 51;

    CameraDeviceType = 52;

    YandexStationChironDeviceType = 53;
    YandexStationPholDeviceType = 54;
}

message TUserDeviceInfo {
    option (NYT.default_field_flags) = SERIALIZATION_YT;

    string Manufacturer = 1 [json_name = "manufacturer", (NYT.column_name) = "manufacturer"];
    string Model = 2 [json_name = "model", (NYT.column_name) = "model"];
    string HwVersion = 3 [json_name = "hw_version", (NYT.column_name) = "hw_version"];
    string SwVersion = 4 [json_name = "sw_version", (NYT.column_name) = "sw_version"];
}

message TUserQuasarInfo {
    option (NYT.default_field_flags) = SERIALIZATION_YT;

    string DeviceId = 1 [json_name = "device_id", (NYT.column_name) = "device_id"];
    string Platform = 2 [json_name = "platform", (NYT.column_name) = "platform"];
}

message TUserSharingInfo {
    option (NYT.default_field_flags) = SERIALIZATION_YT;

    uint64 OwnerID = 1 [json_name = "owner_id", (NYT.column_name) = "owner_id"];
    string HouseholdID = 2 [json_name = "household_id", (NYT.column_name) = "household_id"];
}
